Building on newer systems fails due to using stdargs but not including
the header file:

copydir.c: In function 'error_acl':
copydir.c:126:2: warning: implicit declaration of function 'va_start' [-Wimplicit-function-declaration]
copydir.c:132:2: warning: implicit declaration of function 'va_end' [-Wimplicit-function-declaration]
...
libmisc/copydir.c:126: undefined reference to 'va_start'
libmisc/copydir.c:132: undefined reference to 'va_end'
...
make[2]: *** [useradd] Error 1

Patch authored by Mike Frysinger, taken from Gentoo:

http://sources.gentoo.org/cgi-bin/viewvc.cgi/gentoo-x86/sys-apps/shadow/files/shadow-4.1.5-stdarg.patch

Upstream-Status: Pending

Signed-off-by: Scott Garman <scott.a.garman@intel.com>

diff -urN shadow-4.1.5.orig//libmisc/copydir.c shadow-4.1.5/libmisc/copydir.c
--- shadow-4.1.5.orig//libmisc/copydir.c	2011-12-09 14:23:15.000000000 -0800
+++ shadow-4.1.5/libmisc/copydir.c	2012-04-03 09:05:41.297280296 -0700
@@ -40,6 +40,7 @@
 #include <sys/time.h>
 #include <fcntl.h>
 #include <stdio.h>
+#include <stdarg.h>
 #include "prototypes.h"
 #include "defines.h"
 #ifdef WITH_SELINUX
